File: src/B_1158/B_1158.java
package B_1158;

import java.io.BufferedWriter;
import java.io.IOException;
import java.io.OutputStreamWriter;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class B_1158 {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        sb.append('<');
        int N=sc.nextInt();
        int K=sc.nextInt();
        Queue<Integer> q = new LinkedList<>();
        for(int i=1; i<=N; i++){
            q.offer(i);
        }
        while(q.size() != 1){
            for(int i=0; i<K-1; i++){
                q.offer(q.poll());
            }
            sb.append(q.poll()).append(", ");
        }
        sb.append(q.poll()).append(">");
        bw.write(sb.toString());
        bw.flush();
    }
}

File: src/B_1158/B_1158.kt
package B_1158

import java.io.BufferedWriter
import java.io.OutputStreamWriter
import java.util.*


fun main(args: Array<String>) {
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sc = Scanner(System.`in`)
    val sb = StringBuilder()
    sb.append('<')
    val N = sc.nextInt()
    val K = sc.nextInt()
    val q: Queue<Int> = LinkedList()
    for (i in 1..N) {
        q.offer(i)
    }
    while (q.size != 1) {
        for (i in 0 until K - 1) {
            q.offer(q.poll())
        }
        sb.append(q.poll()).append(", ")
    }
    sb.append(q.poll()).append(">")
    bw.write(sb.toString())
    bw.flush()
}
